e2e/tests/transfer/forwarding_test.go

```diff
@@ -33,6 +33,28 @@ func (s *TransferForwardingTestSuite) SetupSuite() {
 // TestForwarding_WithLastChainBeingICS20v1_Succeeds tests the case where a token is forwarded and successfully
 // received on a destination chain that is on ics20-v1 version.
 func (s *TransferForwardingTestSuite) TestForwarding_WithLastChainBeingICS20v1_Succeeds() {
+	channelBToCCallback := func() ibc.ChannelOutput {
+		ctx := context.TODO()
+		// Creating a new path between chain B and chain C with a ICS20-v1 channel
+		opts := s.TransferChannelOptions()
+		opts.Version = transfertypes.V1
+		chains := s.GetAllChains()
+		channelBtoC, _ := s.CreatePath(ctx, chains[1], chains[2], ibc.DefaultClientOpts(), opts)
+		s.Require().Equal(transfertypes.V1, channelBtoC.Version, "the channel version is not ics20-1")
+		return channelBtoC
+	}
+	s.testForwardingThreeChains(channelBToCCallback)
+}
+
+// TestForwarding_Succeeds tests the case where a token is forwarded and successfully
+// received on a destination chain.
+func (s *TransferForwardingTestSuite) TestForwarding_Succeeds() {
+	s.testForwardingThreeChains(func() ibc.ChannelOutput {
+		return s.GetChainChannel(testsuite.ChainChannelPair{ChainIdx: 1, ChannelIdx: 1})
+	})
+}
+
+func (s *TransferForwardingTestSuite) testForwardingThreeChains(channelBToCCallback func() ibc.ChannelOutput) {
```

Comment:
I feel like callback is more appropriate when you want to execute something based on something that has happened. But this is just an option, happy to go with it:

```suggestion
func (s *TransferForwardingTestSuite) testForwardingThreeChains(createChannelBToC func() ibc.ChannelOutput) {
```

Reply:
It makes sense, though I was talking to Cian on Thursday and it does not actually make sense to over-complicate by passing a function, we can leave the logic in the test (see newest commit)